### PR TITLE
Event Hub: Updated for 2018-Preview

### PR DESCRIPTION
**Pull request by v-Ajnava**
## Description
Swagger Link : https://github.com/Azure/azure-rest-api-specs/pull/3294

Preview supports:
1) Support for Namespace
2) Added support for Network IPFilterRules for namespaces
3) Added support for Virtual NetworkRules for namespaces
4) Added support for Eventhub Cluster

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/EventHub/Management.EventHub/generate.ps1 [hunk 1]
```diff
@@ -1 +1 @@
-Start-AutoRestCodeGeneration -ResourceProvider "eventhub/resource-manager" -AutoRestVersion "latest"
+Start-AutoRestCodeGeneration -ResourceProvider "eventhub/resource-manager" -AutoRestVersion "latest" -ConfigFileTag "package-2018-01-preview"
```
Comment (dsgouda) [inline src/SDKs/EventHub/Management.EventHub/generate.ps1]: nit info: Please make sure that the next time you regenerate the code you apply the correct tags here

File: src/SDKs/EventHub/EventHub.Tests/Tests/ScenarioTests.ConsumergroupsTests.CRUD.cs [hunk 2]
```diff
@@ -1,115 +0,0 @@
...
-    using Microsoft.Rest.Azure;
-    using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
-    using TestHelper;
-    using Xunit;
-    public partial class ScenarioTests
-    {
-        [Fact]
-        public void ConsumerGroupsCreateGetUpdateDelete()
-        {
-            using (MockContext context = MockContext.Start(this.GetType().FullName))
```
Comment (dsgouda) [inline src/SDKs/EventHub/EventHub.Tests/Tests/ScenarioTests.ConsumergroupsTests.CRUD.cs]: Is there a reason you'd like to delete these tests? Are these operations no longer being supported? You can always skip the test if this temporary

**Review by dsgouda — COMMENTED** [review]
Looks great apart from a couple of comments

File: src/SDKs/EventHub/Management.EventHub/generate.ps1 (hunk 1 above)
Comment (v-Ajnava) [reply src/SDKs/EventHub/Management.EventHub/generate.ps1]: Sure Will make note of it.

File: src/SDKs/EventHub/EventHub.Tests/Tests/ScenarioTests.ConsumergroupsTests.CRUD.cs (hunk 2 above)
Comment (v-Ajnava) [reply src/SDKs/EventHub/EventHub.Tests/Tests/ScenarioTests.ConsumergroupsTests.CRUD.cs]: right now, Preview API do not support the operations other than the tests included. so deleted the test. once the Preview and Stable API operations are merged will restore the tests.

Comment (dsgouda) [reply src/SDKs/EventHub/EventHub.Tests/Tests/ScenarioTests.ConsumergroupsTests.CRUD.cs]: In that case, either create a test playlist or skip tests that you wish to restore in the future.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**